### PR TITLE
devdocs-desktop: init at 0.7.1

### PR DESCRIPTION
**Pull request by ymarkus**
Also added ymarkus to maintainers list.

###### Motivation for this change
Add a devdocs-desktop package

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ymarkus** [comment]
Had to add a small gtk fix to fix a minor bug I just noticed

File: pkgs/applications/misc/devdocs-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    substituteInPlace $out/share/applications/devdocs.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A full-featured desktop app for DevDocs.io";
+    longDescription = ''
+      DevDocs.io combines multiple API documentations in a fast, organized, and searchable interface. This is an unofficial desktop app for it.
+    '';
+    homepage = "https://devdocs.egoist.sh";
```
Comment (teto) [inline pkgs/applications/misc/devdocs-desktop/default.nix:37]: got an error on this url ?

Comment (ymarkus) [reply pkgs/applications/misc/devdocs-desktop/default.nix:37]: Yes, it seems as though this url is not working right now, but according to [the source repo](https://github.com/egoist/devdocs-desktop) it is the official homepage for the project.

Do you think I should just put the github repo there?

Comment (teto) [reply pkgs/applications/misc/devdocs-desktop/default.nix:37]: yes, egoist.sh works and seems to be just the blog.

Comment (ymarkus) [reply pkgs/applications/misc/devdocs-desktop/default.nix:37]: Good catch, thanks!

**Comment by ymarkus** [comment]
Rebase: Changed homepage from [broken url](https://devdocs.egoist.sh) to [git repo](https://github.com/egoist/devdocs-desktop).

**Comment by teto** [comment]
dont have much experience with appimageTools packaging so I am hesitant to merge but it works well. I didn't know the tool and it will replace zeal for sure. By any change, do you know if it is possible to consult hackage (haskell website) with it (couldn't find the info)

**Comment by ymarkus** [comment]
Yes, looks like they do have Haskell in their collection. To be more precise, Haskell 7 & 8.

**Pull request merged**